File: LinkedList/python/reverse_linked_list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    head_container = {"head":None}
    def reverse(self,head,curr):
        if curr.next == None:
            self.head_container["head"] = curr
            return
        self.reverse(head=head,curr=curr.next)
        curr.next.next = curr
        curr.next = None 
    def reverseList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        if head==None:
            return head
        self.head_container["head"] = head
        self.reverse(head=self.head_container["head"],curr=self.head_container["head"])
        return self.head_container["head"]

File: LinkedList/typescript/reverse_linked_list.ts

//   Definition for singly-linked list.
  class ListNode {
      val: number
      next: ListNode | null
      constructor(val?: number, next?: ListNode | null) {
          this.val = (val===undefined ? 0 : val)
          this.next = (next===undefined ? null : next)
      }
  }

let head_container:{"head":ListNode|null} = {"head":null}

function reverse(head:ListNode | null,curr:ListNode | null){
    if(curr?.next === null){
        head_container["head"] = curr
        return
    }
    if(curr!=null){
        reverse(head,curr?.next)
        curr.next.next = curr
        curr.next = null
    }
}

function reverseList(head: ListNode | null): ListNode | null {
    head_container["head"] = head
    if (head_container["head"]===null){
        return head
    }
    reverse(head_container["head"],head_container["head"])
    return head_container["head"]
};